src/ctap/hid/mod.rs

```diff
@@ -70,9 +75,40 @@ impl From<u8> for CtapHidCommand {
     }
 }
 
+/// CTAPHID errors
+///
+/// See section 11.2.9.1.6. of FIDO 2.1 (2021-06-15).
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum CtapHidError {
+    /// The command in the request is invalid.
+    InvalidCmd = 0x01,
+    /// A parameter in the request is invalid.
+    _InvalidPar = 0x02,
+    /// The length of a message is too big.
+    InvalidLen = 0x03,
+    /// Expected a continuation packet with a specific sequence number, got another sequence number.
+    ///
+    /// This error code is also used if we expect a continuation packet, and receive an init
+    /// packet. We interpreted it as invalid seq number 0.
+    InvalidSeq = 0x04,
+    /// This packet arrived after a timeout.
+    MsgTimeout = 0x05,
+    /// A packet arrived on one channel while another is busy.
+    ChannelBusy = 0x06,
+    /// Command requires channel lock.
+    _LockRequired = 0x0A,
+    /// The requested channel ID is invalid.
+    InvalidChannel = 0x0B,
+    /// Unspecified error.
+    _Other = 0x7F,
+    /// This error is silently ignored.
+    UnexpectedContinuation,
+}
+
 /// Describes the structure of a parsed HID packet.
 ///
 /// A packet is either an Init or a Continuation packet.
+#[derive(Debug, PartialEq)]
```

Comment:
Is it intended to not derive `Eq`? Looks like equality should be reflexive here.

Reply:
Tests probably rely on `==` to be reflexive which is only true with `Eq`. So technically the tests are wrong even if in practice they're not since the types are morally `Eq`, they just don't advertise it.